File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.py
import sys
from collections import deque
input = sys.stdin.readline

n, m, v = map(int, input().rstrip().split())
graph = [[] for _ in range(n + 1)]

for _ in range(m):
    a, b = map(int, input().rstrip().split())
    graph[a].append(b)
    graph[b].append(a)

for linked in graph:
    linked.sort()

visited = [0] * (n + 1)

def dfs(start):
    visited[start] = 1
    print(start, end=" ")
    for i in graph[start]:
        if not visited[i]:
            dfs(i)


def bfs(start):
    q = deque([start])
    visited[start] = 1
    while q:
        v = q.popleft()
        print(v, end=" ")
        for i in graph[v]:
            if not visited[i]:
                q.append(i)
                visited[i] = 1

dfs(v)
print()
visited = [0] * (n + 1)
bfs(v)

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.cc
#include <iostream>
#include <queue>
using namespace std;

int n, m, v;
int graph[1001][1001];
bool visited[1001];
queue<int> q;

void dfs(int start){
    visited[start] = true;
    cout << start << " ";
    for (int i = 1; i <= n; i++) {
        if (graph[start][i] == 1 && !visited[i]) {
            dfs(i);
        }
    }
}

void bfs(int start){
    q.push(start);
    visited[start] = true;
    while (!q.empty()) {
        int v = q.front();
        q.pop();
        cout << v << " ";
        for (int i = 1; i <= n; i++) {
            if (graph[v][i] == 1 and !visited[i]) {
                visited[i] = true;
                q.push(i);
            }
        }
    }
}

int main(){
    cin >> n >> m >> v;
    
    for (int i = 0; i < m; i++) {
        int a, b;
        cin >> a >> b;
        graph[a][b] = 1;
        graph[b][a] = 1;
    }

    for (int i = 0; i <= n; i++) {
        visited[i] = false;
    }

    dfs(v);

    for (int i = 0; i <= n; i++) {
        visited[i] = false;
    }

    cout << endl;

    bfs(v);

    return 0;
}